### DISCUSSION TITLE
Quality and sequence have different sizes.

### DISCUSSION THREAD

**Discussion by anabteles** [milaboratory/mixcr#1878] [General]
Hi,

As in Izraelson et al. (2017) I am trying to use MIGEC to group my reads by UMI and then MiXCR to extract the repertoires, but I get the following error:

Exception in thread "main" com.milaboratory.core.io.sequence.IllegalFileFormatException: Quality and sequence have different sizes.
	at com.milaboratory.core.io.sequence.fastq.FastqRecordsReader.nextRecord(FastqRecordsReader.java:211)
	at com.milaboratory.core.io.sequence.fastq.SingleFastqReader.guessFormat(SingleFastqReader.java:325)
	at com.milaboratory.core.io.sequence.fastq.SingleFastqReader.<init>(SingleFastqReader.java:257)
	at com.milaboratory.core.io.sequence.fastq.SingleFastqReader.<init>(SingleFastqReader.java:65)
	at com.milaboratory.core.io.sequence.fastq.PairedFastqReader.<init>(PairedFastqReader.java:60)
	at com.milaboratory.mixcr.cli.ActionAlign$AlignParameters.createReader(ActionAlign.java:445)
	at com.milaboratory.mixcr.cli.ActionAlign.go(ActionAlign.java:192)
	at com.milaboratory.cli.JCommanderBasedMain.main(JCommanderBasedMain.java:155)
	at com.milaboratory.mixcr.cli.Main.main(Main.java:115)

Did this happen to you before? Any idea on how to solve the issue or on another approach that I can use?
Many thanks in advance,
Ana

**Comment by anabteles** [comment]
I solved this by changing the format of the fastq file from windows to unix.

**Comment by linqy-immune** [comment]
hello. I encountered the problem like this and I've tried the method you mentioned above, but the error wasn't solved. Could you please share your experience with me? Thank you very much!!!

**Comment by mizraelson** [comment]
Hi, can you please share the command you run, the version of the software and the error you get?

**Comment by linqy-immune** [comment]
Oh, thank you very much!!! The command I ran was "java -Xmx4g -Xms3g -jar C:\mixcr\mixcr.jar align -s hs -p biomed2-human-rna-trbdg "D:\changeSRR14516093.fastq" alignments2.vdjca", the software version was "mixcr4.7.0" and the error log was attached in the file.
[attachment.txt](https://github.com/user-attachments/files/18321176/attachment.txt)

**Comment by mizraelson** [comment]
The issue seems to be related to the file formatting. The FASTQ file you’re working with has Windows-style line breaks (CR+LF, i.e., \r\n) instead of the Unix-style line breaks (LF, i.e., \n) expected by the FASTQ reader.

If you’re using Linux or Mac, you can fix this by running the following command in the terminal:
`sed -i 's/\r$//' filename.fastq`

I also tried downloading the file from https://trace.ncbi.nlm.nih.gov/Traces/sra-reads-be/fastq?acc=SRR14516093 and it worked for me without any issues.

 Avoid opening the file in a Windows text editor, as it might introduce unwanted line breaks.

**Reply by linqy-immune** [reply]
Got it, thank you!!! I'll give it a try.

**Reply by linqy-immune** [reply]
hello, I tried to analyze this group of data just now, the data format is okay, but then I found that almost all the presets available in the mixcr cannot work. Could you please share with me how you ran the  SRR14516093 without any issues. I would appreciate it if you give me some advice.

**Reply by mizraelson** [reply]
What do you mean by “presets cannot work”? I used the `generic-amplicon` preset to check if the file works in general. What protocol was used to generate this data?

**Reply by linqy-immune** [reply]
hello! I ran "analyze" parameter using almost all presets provided in mixcr, but no correct output was exported. The information file of this group of data was "Platform: Ion Torrent, Strategy: Targeted-Capture, Source: TRANSCRIPTOMIC, Selection: PCR, Layout: Single", please have a look. Thanks a lot!

**Reply by linqy-immune** [reply]
hello! I attach my file here, could you please help me have a look? I am a big fan of mixcr and really wanna analyze the data by mixcr but not other softwares. Thank you!
[no alignment.txt](https://github.com/user-attachments/files/18548850/no.alignment.txt)

**Reply by mizraelson** [reply]
Answered in #1895
